From 4532da5257a2af010c67776c42ab08417d713d79 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 11 Dec 2024 10:04:48 +0200
Subject: [PATCH] [FSW-9022] fs1: address_space: fix pcie phy defines

IO0_PCIE_FW_CSR_BASE definition is defined now without
HL_CFG_BASE included, hence when used with memcpy,
HL_CFG_BASE should be added.

Change-Id: I450058c3c94e04a6f06399678a2a9eafe68548ab
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/arch/fs1/arc/palad.c b/arch/fs1/arc/palad.c
index e4daf6c..e70fa22 100644
--- a/arch/fs1/arc/palad.c
+++ b/arch/fs1/arc/palad.c
@@ -255,6 +255,7 @@
 void hl_pcie_phy_fw_copy(void)
 {
 	uint64_t full_addr;
+	uint8_t *dst_csr = (uint8_t *)IO0_PCIE_FW_CSR_BASE + HL_CFG_BASE;
 	/* sram_loc_addr located at PPBOOT_LAST_ADDR, aligned to 16 bytes */
 	uint32_t sram_loc_addr = ALIGN_TO_16_BYTES(PPBOOT_LAST_ADDR);
 	const uint32_t *csr_loc_data = get_csr_locations();
@@ -271,7 +272,7 @@
 		(uint32_t)PHY_FW_SIZE, false, false);
 
 	/* Copy CSR_LOC_DATA to IO0_PCIE_FW_CSR_BASE */
-	memcpy((void *)IO0_PCIE_FW_CSR_BASE, (const void *)csr_loc_data, get_csr_locations_data_size());
+	memcpy(dst_csr, (const void *)csr_loc_data, get_csr_locations_data_size());
 }
 
 static void palad_pcie_params_init(void)
